replica/table.cc

```diff
@@ -306,11 +319,14 @@ table::make_streaming_reader(schema_ptr s, reader_permit permit,
         return make_combined_reader(s, std::move(permit), std::move(readers), fwd, fwd_mr);
     });
 
-    return make_flat_multi_range_reader(s, std::move(permit), std::move(source), ranges, slice, nullptr, mutation_reader::forwarding::no);
+    return maybe_compact_for_streaming(
+            make_flat_multi_range_reader(s, std::move(permit), std::move(source), ranges, slice, nullptr, mutation_reader::forwarding::no),
+            get_compaction_manager(),
+            compaction_time);
 }
 
 flat_mutation_reader_v2 table::make_streaming_reader(schema_ptr schema, reader_permit permit, const dht::partition_range& range,
-        const query::partition_slice& slice, mutation_reader::forwarding fwd_mr) const {
+        const query::partition_slice& slice, mutation_reader::forwarding fwd_mr, std::optional<gc_clock::time_point> compaction_time) const {
```

Comment:
We should make this overload call the other.

Reply:
The returned reader out-lives the co-routine frame.